v2/examples/spiffe-grpc/README.md

```diff
@@ -0,0 +1,101 @@
+# gRPC SVID example 
+
+This example shows how two services using gRPC can communicate using mTLS with X509 SVIDs obtained from SPIFFE workload API.
+
+Each service is connecting to workload API to fetch its identities. Since this example assumes the SPIRE implementation, it uses the SPIRE default socket path: `/tmp/agent.sock`. 
+
+```go
+source, err := workloadapi.NewX509Source(ctx, workloadapi.WithClientOptions(workloadapi.WithAddr(socketPath)))
+```
+
+In case socket path is not provided using [SourceOption](../../workloadapi/option.go#L39) the value from the environment variable `SPIFFE_ENDPOINT_SOCKET` is used
+
+```go
+source, err := workloadapi.NewX509Source(ctx)
+```
+
+Then, the **gRPC server** creates a `tls.Config` with server configurations to allow mTLS connections, using previously created [workloadapi.X509Source](../../workloadapi/x509source.go#L17) and validates than the certificate presented to server from client has SPIFFE ID `spiffe://examples.org/client`.
```

Comment:
The relative links to links in the source code are destined to become stale. Maybe we should link to the go package documentation instead:
https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/workloadapi?tab=doc#X509Source